### ISSUE TITLE
Model displayed incorrectly under Ubuntu 18.04

### ISSUE THREAD

**Issue by maximilianmuehlbauer** [colmap/colmap#766]
**Describe the bug**
Colmap reconstructions are displayed incorrectly in the GUI under Ubuntu 18.04. Cameras and points are not displayed in their correct places but at seemingly random positions.

**To Reproduce**
1. Download the RobotCar Seasons dataset: https://drive.google.com/drive/folders/1tNZcxSL9GtLBx10Bmgg1qSEnOiRtFcO5
2. Open 001_aligned (or any of the others) from 3D-models/individual/colmap_reconstructions
3. Cameras and Points are displayed wrong (see images attached)

**Expected behavior**
Display the correct camera poses and points like under Windows.

**Screenshots**
Model under Windows:
![windows](https://user-images.githubusercontent.com/2041080/70930579-d4934c00-2035-11ea-812c-a5230f31ec36.png)

Model under Ubuntu 18.04:
![colmap_linux](https://user-images.githubusercontent.com/2041080/70930588-d9f09680-2035-11ea-95af-4bedaf847037.png)

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04
 - Version: commit f4eaade
- Compiled using the installation instructions of the documentation

**Comment by tsattler** [comment]
Just tried it out and I cannot reproduce the error. For me, the model is displayed correctly.
My system is Ubuntu 18.04 and the Colmap version is 0dce1db. Looking at the commit history in between 0dce1db and f4eaade, I can't seem anything that could have broken things.

One potential issues might be the graphics driver, i.e., that the GPU is not able to handler larger coordinates.

**Comment by maximilianmuehlbauer** [comment]
This actually happens on the same machine; under Linux the nvidia driver 440 is installed and under Windows 441 on a GTX 1050 Ti (notebook) but it also happens on a GTX 1060 (desktop) also with driver 440. Is there any way to check if there's something wrong with the graphics driver? Are there other packages involved I could check?

**Comment by tsattler** [comment]
Do graphics-heavy application work for you on Ubuntu? Is Colmap able to use the GPU for feature extraction and matching?
To me, the visualization looks like a failure to handle too large coordinates as there seems to be a lot of quantization / rounding going on..

**Comment by maximilianmuehlbauer** [comment]
Deep Learning with pytorch (all installed in a conda environment so there are hopefully no inferences) works well. Feature extraction and matching seems to at least put a (high) workload on the GPU. Superficially inspecting a few of the matches also lets me think they seem to be reasonable. Rotating the model quickly also increases GPU load so I think colmap somehow definitely uses the GPU.

Could CUDA 10.2 be an issue? Is there any other application that uses comparable code I could test? Or is there some other way to check if quantization / rounding causes the problem?

**Comment by ahojnnes** [comment]
This problem sometimes happens when you use COLMAP text files in combination with a different system locale (I guess German in your case), where commas and dots have the reverse meaning. In this case, COLMAP interprets numbers completely differently. The solution would be to either use the binary file format or change your system locale to English.

**Comment by ahojnnes** [comment]
@tsattler You might want to ship the models using binary files. Should be as simple as running the `model_converter` command.

**Comment by maximilianmuehlbauer** [comment]
@ahojnnes thank you very much that solved the issue!

Executing `export LC_NUMERIC="en_US.UTF-8"` before opening colmap makes the model look correctly.

Do you know if the locale setting also influences other parts of colmap?

**Comment by SBCV** [comment]
Would it be possible to fix the txt-model parsing functionality of colmap instead of exporting `LC_NUMERIC`? 
I exported some results as txt and re-imported it, when observing this weird behavior. I was about to debug the corresponding code, when I stumbled more or less accidentally over this thread. 

I think nobody expects to change a system locale variable to make the export / import functionality to work.
 
I can provide a PR if needed. Just let me know if this could be fixed with a reasonable number of changes - or if this is something fundamental (which can not be easily fixed).
